File: 0273_Modest - 2013 - Radiative heat transfer/代码/第四版/Chapters本书/Chapter02-03/fresnel.m
function [RHOS, RHOP, RHO] = fresnel(AN, AK, TH)  

C1 = AN^2 - AK^2 - (sin(TH))^2;
P2 = .5E0*(sqrt(C1^2 + (2.E0*AN*AK)^2) + C1);
Q2 = P2 - C1;
CT = cos(TH);
C2 = (AN^2-AK^2)*CT;
C3 = 2.E0*AN*AK*CT;
P  = sqrt(P2);
Q  = sqrt(Q2);
RHOP = ((P-C2)^2 + (Q-C3)^2)/((P+C2)^2 + (Q+C3)^2);
RHOS = ((CT-P)^2 + Q2)/((CT+P)^2 + Q2);
RHO  =  .5E0*(RHOP + RHOS);

File: 0273_Modest - 2013 - Radiative heat transfer/代码/第三版/chapter_02_03/fresnel.cpp
// Frensel Function

# include <iostream.h>
# include <math.h>


	
void FRESNEL(double AN, double AK, double TH, double& RHOS, double& RHOP,
				double& RHO)
{
	double C1,P2,Q2,CT,C2,C3,P,Q;

	C1=AN*AN-AK*AK-pow(sin(TH),2);
	P2=.5E0*(sqrt(C1*C1+pow((2.E0*AN*AK),2))+C1);
	Q2=P2-C1;
	CT=cos(TH);
	C2=(AN*AN-AK*AK)*CT;
	C3=2.E0*AN*AK*CT;
	P=sqrt(P2);
	Q=sqrt(Q2);
	RHOP=(pow((P-C2),2)+pow((Q-C3),2))/(pow((P+C2),2)+pow((Q+C3),2));
	RHOS=(pow((CT-P),2)+Q2)/(pow((CT+P),2)+Q2);
	RHO=.5E0*(RHOP+RHOS);
}
